torch/csrc/api/include/torch/serialize/input-archive.h

```diff
@@ -44,6 +44,12 @@ class TORCH_API InputArchive final {
 
   ~InputArchive() = default;
 
+  /// Reads a `tensor` associated with a given `key`. If the `InputArchive` doesn't
```

Comment:
This API seems pretty awkward.  For example, what is the expected behavior if a tensor with `key` exists but I pass in the wrong value for `is_buffer`?  (Why do I have to get the right anyway?).

The reason this API exists seems questionable -- instead of serializing each element to a unique key, why don't I serialize the entire vector to a single key and have the serialization format specify the number of entries?  This is what is already done elsewhere in the code.  Consider:
https://github.com/pytorch/pytorch/blob/de582e2f89d154ef7579caf8fe8aac80c2d8a569/torch/csrc/api/include/torch/optim/serialize.h#L32-L60